src/core/Button/Button.js

```diff
@@ -3,27 +3,17 @@ import PropTypes from 'prop-types'
 import Icon from '../Icon'
 import s from './styles'
 
-function Button({
-    role,
-    type,
-    size,
-    kind,
-    icon,
-    label,
-    active,
-    disabled,
-    onClick,
-}) {
+function Button({ role, type, kind, icon, label, active, disabled, onClick }) {
     return (
         <button
             type={type}
             role={role}
             disabled={disabled}
             onClick={onClick}
-            className={s('button', kind, size, { active })}
+            className={s('button', kind, { active })}
         >
             {icon && <Icon name={icon} />}
-            {label}
+            {label && <span>{label}</span>}
```

Comment:
do we need a span here, as we had it working without it before

Reply:
We were using a helper before called something like `wrapTextNodeInSpan`. The problem is not with the text itself but with the icon. If there is no text, the icon needs to be aligned centrally, without margins. I achieve this with this selector `.button i:only-child`. Unfortunately the `:only-child` selector doesn't take text nodes into account. I don't see how we can achieve good alignment without. Any advice is welcome.